File: frontend/types/dataset.ts
export interface ColumnInfo {
  name: string;
  type: string;
  nullable: boolean;
}

export interface TableInfo {
  name: string;
  row_count?: number;
  columns: ColumnInfo[];
}

export interface Dataset {
  id: string;
  name: string;
  source: "upload" | "external_db";
  created_at: string;
  tables: TableInfo[];
}

export interface UploadResponse {
  success: boolean;
  dataset_id: string;
  table_name: string;
  row_count: number;
  columns: ColumnInfo[];
}

export interface ConnectDbRequest {
  name: string;
  host: string;
  port: number;
  database: string;
  username: string;
  password: string;
  ssl: boolean;
}

export interface ConnectDbResponse {
  success: boolean;
  dataset_id: string;
  tables: TableInfo[];
}

File: backend/app/schemas/dataset.py
"""
backend/app/schemas/dataset.py
Pydantic schemas for dataset endpoints.
"""
from typing import List, Optional
from pydantic import BaseModel, Field


class ColumnSchema(BaseModel):
    name: str
    type: str
    nullable: bool = True


class TableSchema(BaseModel):
    name: str
    row_count: Optional[int] = 0
    columns: List[ColumnSchema]


class DatasetResponse(BaseModel):
    id: str
    name: str
    source: str
    table_name: str
    row_count: int
    created_at: Optional[str] = None
    tables: List[TableSchema]


class UploadResponse(BaseModel):
    success: bool = True
    dataset_id: str
    table_name: str
    row_count: int
    columns: List[ColumnSchema]
